temporal-serviceclient/src/main/java/io/temporal/serviceclient/ChannelManager.java

```diff
@@ -0,0 +1,341 @@
+      for (ClientInterceptor interceptor : grpcClientInterceptors) {
+        channel = ClientInterceptors.intercept(channel, interceptor);
+      }
+    }
+
+    // should be after grpcClientInterceptors to be closer to the head and to let the
+    // grpcClientInterceptors
+    // observe requests with grpcClientInterceptors already set
+    Collection<GrpcMetadataProvider> grpcMetadataProviders = options.getGrpcMetadataProviders();
+    if (grpcMetadataProviders != null && !grpcMetadataProviders.isEmpty()) {
+      GrpcMetadataProviderInterceptor grpcMetadataProviderInterceptor =
+          new GrpcMetadataProviderInterceptor(grpcMetadataProviders);
+      channel = ClientInterceptors.intercept(channel, grpcMetadataProviderInterceptor);
+    }
+    return channel;
+  }
+
+  private ManagedChannel prepareChannel() {
+    NettyChannelBuilder builder =
+        NettyChannelBuilder.forTarget(options.getTarget())
+            .defaultLoadBalancingPolicy("round_robin")
+            .maxInboundMessageSize(MAX_INBOUND_MESSAGE_SIZE);
+    if (options.getEnableKeepAlive()) {
+      builder
+          .keepAliveTime(options.getKeepAliveTime().toMillis(), TimeUnit.MILLISECONDS)
+          .keepAliveTimeout(options.getKeepAliveTimeout().toMillis(), TimeUnit.MILLISECONDS)
+          .keepAliveWithoutCalls(options.getKeepAlivePermitWithoutStream());
+    }
+
+    if (options.getSslContext() == null && !options.getEnableHttps()) {
+      builder.usePlaintext();
+    } else if (options.getSslContext() != null) {
+      builder.sslContext(options.getSslContext());
+    } else {
+      builder.useTransportSecurity();
+    }
+
+    // Disable built-in idleTimer until https://github.com/grpc/grpc-java/issues/8714 is resolved.
+    // jsdk force-idles channels often anyway, so this is not needed until we stop doing
+    // force-idling as a part of
+    // https://github.com/temporalio/sdk-java/issues/888
+
+    // Why 31 days? See ManagedChannelImplBuilder#IDLE_MODE_MAX_TIMEOUT_DAYS and
+    // https://github.com/grpc/grpc-java/issues/8714#issuecomment-974389414
+    builder.idleTimeout(31, TimeUnit.DAYS);
+
+    if (options.getChannelInitializer() != null) {
+      options.getChannelInitializer().initChannel(builder);
+    }
+
+    return builder.build();
+  }
+
+  private void initConnectionManagement() {
+    // Currently, it is impossible to modify backoff policy on NettyChannelBuilder.
+    // For this reason we reset connection backoff every few seconds in order to limit maximum
+    // retry interval, which by default equals to 2 minutes.
+    // Once https://github.com/grpc/grpc-java/issues/7456 is done we should be able to define
+    // custom policy during channel creation and get rid of the code below.
+    if (options.getConnectionBackoffResetFrequency() != null) {
+      grpcConnectionManager.scheduleWithFixedDelay(
+          resetGrpcConnectionBackoffTask(),
+          options.getConnectionBackoffResetFrequency().getSeconds(),
```

Comment:
There is a lot of `getSeconds()` here and health check timeout and such. Can we use the more granular unit, e.g. milliseconds?